lib/src/minio.dart

```diff
@@ -67,16 +67,34 @@ class Minio {
   final _regionMap = <String, String>{};
 
   /// Checks if a bucket exists.
+  ///
+  /// Returns `true` if the [bucket] exists or you don't have permission to access
+  /// it (which implies that it exists). Returns `false` only if the [bucket] does
+  /// not exist.
   Future<bool> bucketExists(String bucket) async {
     MinioInvalidBucketNameError.check(bucket);
     try {
-      await _client.request(method: 'HEAD', bucket: bucket);
+      final response = await _client.request(method: 'HEAD', bucket: bucket);
+      validate(response);
+      return (response.statusCode == 200 || response.statusCode == 403);
```

Comment:
`validate(response) throws MinioS3Error when response.statusCode >= 400, so `response.statusCode == 403` will always evaluate to `false`.

Reply:
Ooops, you're right about this. I changed the return value accordingly. Thank you for your comment!